### ISSUE TITLE
Error 500 during setupDatabase Croogo-3.0.5

### ISSUE THREAD

**Issue by muaz234** [croogo/croogo#891]
Greetings,

I have an issue that i cannot proceed after installation of setupDatabase() on Croogo-3.0.5
```
Solution i have tried: 
1) install php7.2 - mbstring PHP extension
2) install php7.2 - intl PHP extension
3) install php7.2 - simplexml PHP extension
4) do manual, web installer and composer installer method.all return same result of error 500 after setupDatabase
```
this is my error log of croogo 3.0.5 skeleton app. It is returning me a blank screen which written 500 Internal Server Error
![screenshot from 2019-02-09 17-28-25](https://user-images.githubusercontent.com/36180180/52519041-67ce5900-2c90-11e9-884f-4c64e2c8264e.png)

do provide me some advice on how to overcome this issue in Croogo 3.0.5 installation. Previously i was using croogo 2.3.2(CakePHP 2) and now this is my first try to learn CakePHP 3. My machine is running on Ubuntu 18 and using Apache web server with php 7.2. Been searching in the forum but none matches to my problem and solution tested and resulting same errror.

I appreciate your suggestions. Thanks

**Comment by rchavik** [comment]
Seems that it cannot write to error.log file.  Otherwise, it should have more info there. If you have solved it by changing selinux setting, what is the content of error.log file?

Another option to try, Instead of using apache to serve it during install, try using php web server, eg:

```sh
cd /var/www/html/cake3 # assuming this is where you have `croogo/app`
bin/cake server  # then browse to http://localhost:8765
```

Also see https://github.com/croogo/croogo/issues/850#issuecomment-386802907, maybe it can serve as temporary workaround

**Comment by muaz234** [comment]
updated the post and 

> Seems that it cannot write to error.log file. Otherwise, it should have more info there. If you have solved it by changing selinux setting, what is the content of error.log file?
> 
> Another option to try, Instead of using apache to serve it during install, try using php web server, eg:
> 
> ```shell
> cd /var/www/html/cake3 # assuming this is where you have `croogo/app`
> bin/cake server  # then browse to http://localhost:8765
> ```
> 
> Also see [#850 (comment)](https://github.com/croogo/croogo/issues/850#issuecomment-386802907), maybe it can serve as temporary workaround

tested on your solution but none works..still returning same error. I have updated the post.

**Comment by rchavik** [comment]
Seems like the error is now different.

Can you try apply the following:
```diff
diff --git a/Core/src/Template/Layout/error.ctp b/Core/src/Template/Layout/error.ctp
index 60750498..16e76d4e 100644
--- a/Core/src/Template/Layout/error.ctp
+++ b/Core/src/Template/Layout/error.ctp
@@ -11,6 +11,10 @@ use Cake\Core\Configure;
 $siteTitle = Configure::read('Site.title');
 $siteTagline = Configure::read('Site.tagline');
 
+$this->loadHelper('Croogo/Core.Layout');
+$this->loadHelper('Croogo/Core.Theme');
+$this->loadHelper('Croogo/Menus.Menus');
+
 ?>
 <!DOCTYPE html>
 <html style='background: #0e0e0e;'>
diff --git a/Install/src/Template/Layout/install.ctp b/Install/src/Template/Layout/install.ctp
index dac6f74d..a7227a87 100644
--- a/Install/src/Template/Layout/install.ctp
+++ b/Install/src/Template/Layout/install.ctp
@@ -35,6 +35,7 @@
                 </h3>
                 <div class="card-body">
                     <?php
+                    $this->loadHelper('Croogo/Core.Layout');
                     echo $this->element('installer_steps');
                     echo $this->Layout->sessionFlash();
                     echo $this->fetch('content');
```

If you have keys too big error during install, try using charset utf8 with utf8_general_ci and see if it helps

**Comment by hacheraw** [comment]
Same error here.
Solved by changing database collation to utf8_general_ci

**Comment by rchavik** [comment]
Solution for keys too big error: https://github.com/croogo/croogo/issues/850#issuecomment-469984266

**Comment by muaz234** [comment]
Hello there, sorry for late reply. At the moment, i am still hitting 500 Internal Server Error prior to installation. I have tried two approach which is extract to /var/www/html/ folder and install from web based installer but still fails. I attach the error log i got while installing it.
![Screenshot from 2019-03-11 22-04-02](https://user-images.githubusercontent.com/36180180/54132393-1eb82300-444f-11e9-9650-080c8777f840.png)

Another approach, i have tried to use composer to install with `--prefer-source` flag and resulted to 500 Internal Server Error with some longer error report details on my error.log while it is installing.
![Screenshot from 2019-03-11 22-40-07](https://user-images.githubusercontent.com/36180180/54132512-4f985800-444f-11e9-952d-17185aec6a5d.png)





![Screenshot from 2019-03-11 22-40-51](https://user-images.githubusercontent.com/36180180/54132530-5921c000-444f-11e9-94ef-8b2a05f906af.png)
![Screenshot from 2019-03-11 22-40-43](https://user-images.githubusercontent.com/36180180/54132531-5921c000-444f-11e9-98c8-f38edc011447.png)
![Screenshot from 2019-03-11 22-40-32](https://user-images.githubusercontent.com/36180180/54132532-59ba5680-444f-11e9-84c3-01f6e5c520f1.png)
![Screenshot from 2019-03-11 22-40-26](https://user-images.githubusercontent.com/36180180/54132533-59ba5680-444f-11e9-9da2-75c911853e4d.png)
![Screenshot from 2019-03-11 22-40-18](https://user-images.githubusercontent.com/36180180/54132534-5a52ed00-444f-11e9-90b4-250d7be1ae39.png)



For both installation I am using a UTF8 charset with utf8_general_ci . Please advice me if any of you experienced this before. Hopefully, there will be resolution to my problem.

**Comment by rchavik** [comment]
Should be fixed in 3.0.6
